src/Fabulous.AST.Tests/TypeDefinitions/Record.fs

```diff
@@ -0,0 +1,76 @@
+﻿namespace Fabulous.AST.Tests.TypeDefinitions
+
+open FSharp.Compiler.Text
+open Fabulous.AST.Tests
+open Fantomas.Core.SyntaxOak
+open NUnit.Framework
+
+open Fabulous.AST
+open type Ast
+
+module Record =
+    [<Test>]
+    let ``Produces a record`` () =
+
+        AnonymousModule() {
+            Record("Colors") {
+                Field("Red", Type.FromString "int")
```

Comment:
What is the most natural way regarding passing different kind of `Type`?
Should we prefer the use of helper functions like here?

Initially, we wanted to provide different widgets that would take care of the conversion to Type, based on the use case: `Field`, `TupleField`, etc

Eg.
```fs
Record("Colors") {
    Field("Red", "int")
    TupleField("Blue") {
         Field("r", "byte")
         Field("g", "byte")
         Field("b", "byte")
    }
    FuncField("ExpensiveGreen", ...)
}
```

Reply:
So, named tuple isn't possible in F# records. ([sample](https://fsprojects.github.io/fantomas-tools/#/ast?data=N4KABGBEAmCmBmBLAdrAzpAXFSAacUiaAYmolmPAIYA2as%2BEkAxgPZwWQAuAngA6wwAYVY1WAJwAiAVTABeADriFyCBAA%2BYAEI0AroNbww4zACMeXQQCowAczMXrYUw8tKV75LwHDREtPKeamDAQcFqOvrYJuaWYDb2sU4uSWEQAL6QIOlAA)).

And this approach feel wrong to me.
I'm not a fan of re-using `Field` for both DU and Record to start with. 

They have different rules altogether and only would work to share for the very basic minimal representation.
You can add things to RecordFields that you can't to UnionCaseField, so I would split those up sooner than later.

As for types, they are their own thing and one of the major kinds of AST nodes. So I won't try to merge them with any given situation where you would use them.

I'm a bit concerned that accepting any string as a type will lead user to abuse them.

I was thinking something along these lines makes sense:

```fsharp
Record("Colors") {
    Field("Red", Type.FromString "int")
    Field(
        "Blue,
        // byte * byte * byte
        TupleType () {
             Type.FromString "byte"
             Type.FromString "byte"
             Type.FromString "byte"
        }
    )
    Field(
        "Green",
        // int -> int -> int
        FuncType () { 
             Type.FromString "int"
             Type.FromString "int"
             Type.FromString "int"
        }
    )
}
```

Reply:
This is quite interesting 🤔. I think we need to experiment with your approach and try to enforce some generated allowed by the F# compiler. But to a certain level because otherwise this can really difficult to model in the DSL